### PR TITLE
92732 frontend play send alle put post til main api

### PR DESCRIPTION
**Pull request by arildbjerkenes**
[AB#96019](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/96019)
[AB#92732](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/92732)
[AB#92737](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/92737)

### CODE REVIEW

File: src/offline/UpdateHandlers/utils.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ export const getPunchTypeById = async (
  * We will get a floating number between 0 and 1. This should ensure that we don't get an id already in use.
```
Comment (VRiveland) [inline src/offline/UpdateHandlers/utils.ts:118]: Change comment?

File: src/offline/syncUpdatesWithBackend.ts [hunk 2]
```diff
@@ -2,28 +2,175 @@ import { ProcosysApiService } from '../services/procosysApi';
...
-                api.postAttachmentByFetch(
-                    updateRequest.url,
-                    updateRequest.bodyData
+
+    const offlineUpdates =
+        await offlineUpdateRepository.getRequestsOrderedByEntityAndSeq();
+
+    //Get a distinct set of entities to syncronize
+    const entitiesToUpdate = new Set<number>();
+    for (const offlineUpdate of offlineUpdates) {
```
Comment (VRiveland) [inline src/offline/syncUpdatesWithBackend.ts:142]: We usually avoid for loops as often as possible. Try to use offlineUpdates.forEact or another array method

File: src/offline/syncUpdatesWithBackend.ts [hunk 3]
```diff
@@ -2,28 +2,175 @@ import { ProcosysApiService } from '../services/procosysApi';
...
+        entitiesToUpdate.add(offlineUpdate.entityid ?? 0);
+    }
+
+    for (const entityIdToUpdate of entitiesToUpdate) {
```
Comment (VRiveland) [inline src/offline/syncUpdatesWithBackend.ts:146]: Same as above, Set also has forEach method

File: src/offline/syncUpdatesWithBackend.ts [hunk 4]
```diff
@@ -2,28 +2,175 @@ import { ProcosysApiService } from '../services/procosysApi';
...
+        //Syncronize updates for the specific entity
+
+        //Find all updates for the current entity
+        const updatesForEntity = offlineUpdates.filter(
+            (offlineUpdate) => offlineUpdate.entityid == entityIdToUpdate
+        );
+
+        let newEntityId;
+
+        for (const offlineUpdate of updatesForEntity) {
```
Comment (VRiveland) [inline src/offline/syncUpdatesWithBackend.ts:156]: Same as above

**Comment by VRiveland** [comment]
Scope list isn't updated when a checklist is changed offline. 
image
Before signing online
image
After signing bottom checklist online

**Comment by VRiveland** [comment]
Localhost doesn't work after sync. After refresh I can't use localhost until I restart the app. 
image
I can't access the app at all

File: src/offline/UpdateHandlers/handleAttachment.ts [hunk 5]
```diff
@@ -49,13 +51,14 @@ const addNewAttachment = async (
...
-    bodyData: any
+    offlinePostRequest: OfflineUpdateRequest,
+    params: URLSearchParams
 ): Promise<void> => {
     const plantId = params.get('plantId');
     const title = params.get('title');
     const punchIdStr = params.get('punchItemId');
-    const blob: Blob = bodyData[0];
+    //const blob: Blob = offlinePostRequest.bodyData[0];
+    const arrayBuffer = offlinePostRequest.blob; //await blob.arrayBuffer();
```
[the remark this answers is not in the document]
Comment (arildbjerkenes) [reply src/offline/UpdateHandlers/handleAttachment.ts:61]: We have to change this to be able to post new attachments. When we find a way to get this to work, we can implement this the same way for WO.

**Pull request merged**